src/java/org/apache/cassandra/cql3/selection/Selectable.java

```diff
@@ -245,18 +266,20 @@ public Selector.Factory newSelectorFactory(TableMetadata table,
             if (column.isPrimaryKeyColumn())
                 throw new InvalidRequestException(
                         String.format("Cannot use selection function %s on PRIMARY KEY part %s",
-                                      isWritetime ? "writeTime" : "ttl",
+                                      kind.name,
                                       column.name));
-            if (column.type.isCollection())
+
+            // maxwritetime is allowed for collection
+            if (kind != Kind.MAX_WRITE_TIME && column.type.isCollection())
```

Comment:
we could potentially move this to the enum? `if (column.type.isCollection() && !kind.allowedForCollection()) then throw...`. If we ever add additional `Kind`s, we wouldn't need to touch this logic, so it becomes a little more maintainable